### DISCUSSION TITLE
TornadoVM and OpenJDK Babylon project

### DISCUSSION THREAD

**Discussion by andrii0lomakin** [beehive-lab/TornadoVM#394] [Q&A] [answered]
Hi.

Is there any collaboration between OpenJDK Babylon and the TornadoVM project? 
If so, to which extent? Is it planning to create integration with code reflection API? Are you participating in spec creation/development?

**Comment by kotselidis** [comment, marked as answer]
Hallo,

We follow the developments within project Babylon and our plan is to be compatible with the code reflection API.
In fact Babylon is using our SPIR-V code gen.
The level of integration is not decided yet as we learn more and more about Babylon as code is being released and we understand it better.
A first level of integration could be the enablement of code derived by Babylon and HAT to be interoperable with our Graal-based code gen under the TornadoVM runtime.
Another could be the usage of Babylon to implement parts of TornadoVM.
Again nothing has been decided at this point as we follow the development.
